codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -168,9 +168,7 @@ private int unconsumedBytes(Http2Stream stream) {
 
     void onGoAwayRead0(ChannelHandlerContext ctx, int lastStreamId, long errorCode, ByteBuf debugData)
             throws Http2Exception {
-        // Don't allow any more connections to be created.
         connection.goAwayReceived(lastStreamId, errorCode, debugData);
```

Comment:
@Scottmitch @nmittler One issue I am wrestling with is also in what order to notify the two listeners. I would argue the connection listeners should execute after the frame listener, because the connection listener actually modifies stream states and there is really no reason why those two listeners should be executed in a different state? Also maybe we should move the logic that closes streams > lastStreamId out of `goAwayReceived(...)`?